lib/puppet/provider/user/useradd.rb

```diff
@@ -94,11 +94,10 @@ def shell=(value)
     value !~ /\s/
   end
 
-  has_features :manages_homedir, :allows_duplicates, :manages_expiry
+  has_features :allows_duplicates, :manages_expiry, :manages_shell
+  has_features :manages_homedir unless 'Solaris'.eql? Facter.value(:operatingsystem)
```

Comment:
Thanks for your contribution @pillarsdotnet! I find `unless 'Solaris' == Factor.value(:operatingsystem)` or `if 'Solaris' != ...` easier to read.

Could you update the commit description about what the new behavior is if you try to remove a user and `managehome => true`? Does puppet fail to apply the catalog or does it silently ignore the `managehome` parameter? I assume it's the latter?

Reply:
I don't have access to a Solaris VM for testing.

Reply:
It doesn't look like much has changed.
```
root@hk6zbqcg7tncoam:~# grep has_features /opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/provider/user/useradd.rb
  has_features :allows_duplicates, :manages_expiry
  has_features :manages_homedir unless 'Solaris' == Facter.value(:operatingsystem)
  has_features :system_users unless %w{HP-UX Solaris}.include? Facter.value(:operatingsystem)
  has_features :manages_passwords, :manages_password_age if Puppet.features.libshadow?
  has_features :manages_shell
root@hk6zbqcg7tncoam:~# /opt/puppetlabs/bin/puppet apply -e 'user { "testuser": ensure => "present", managehome => false }'
Notice: Compiled catalog for hk6zbqcg7tncoam.delivery.puppetlabs.net in environment production in 0.01 seconds
Notice: /Stage[main]/Main/User[testuser]/ensure: created
Notice: Applied catalog in 0.04 seconds
root@hk6zbqcg7tncoam:~# /opt/puppetlabs/bin/puppet apply -e 'user { "testuser": ensure => "absent",  managehome => true  }'
Notice: Compiled catalog for hk6zbqcg7tncoam.delivery.puppetlabs.net in environment production in 0.01 seconds
Error: Could not delete user testuser: Execution of '/usr/sbin/userdel -r testuser' returned 12: UX: /usr/sbin/userdel: ERROR: Unable to remove home directory. No such file or directory
Error: /Stage[main]/Main/User[testuser]/ensure: change from 'present' to 'absent' failed: Could not delete user testuser: Execution of '/usr/sbin/userdel -r testuser' returned 12: UX: /usr/sbin/userdel: ERROR: Unable to remove home directory. No such file or directory
Notice: Applied catalog in 0.05 seconds

```

Reply:
Well, that sucks.  Guess the underlying implementation has to do something like:
```
if `Solaris` == Facter.value(:operatingsystem) {
  homedir = gethomedir(username)
  if file.absent?(homedir) {
    manageuser = false
  }
}
```

Seems incredibly hackish, but I'll work on it.